frame/staking/src/benchmarking.rs

```diff
@@ -477,6 +405,136 @@ benchmarks! {
 			call.dispatch(RawOrigin::Signed(caller.clone()).into())?;
 		}
 	}
+
+	submit_solution_initial {
+		let v in 1 .. 10;
+		let n in 1 .. 100;
```

Comment:
you always benchmark for v in 1..10 and n in 1..100 but there can be way more validator and nominator in the snapshots. Maybe we should benchmark the cost of decoding the Snapshot independently.

I worry the coefficient 35us found here is more the cost of checking the winners than actually decoding SnapshotValidators. (and same for nominators)

I think it would be cool to distinguish 4 cost:
* number of validators (cost associated for querying and checking with ValidatorSnapshot)
* number of nominators (cost associated for querying and checking with NominatorSnapshot)
* number of winner in the solution
* number of assignment (or number of nominator considering they all have maximum assignment) in the solution.

This way I expect to find way smaller coefficient in front of totalNumberOfValidator and totalNumberOfNominator

Reply:
> you always benchmark for v in 1..10 and n in 1..100 but there can be way more validator and nominator in the snapshots. Maybe we should benchmark the cost of decoding the Snapshot independently.

The size of the benchmarks is exactly the same as n and v here, so this is not correct. 

The rest of your comment is indeed a great point: I am probably ignoring the size of the solution in the `base weight` part. in the benchmarks we create `v` validators, and we elect them all. This is too simplistic. There should be benchmarks which has the 4 elements that you mentioned.

Reply:
Trimming the size of a phragmen solution will be hard here, I need either probably more time here, or someone's help.

Reply:
>  The size of the benchmarks is exactly the same as n and v here, so this is not correct.

What is size of benchmarks ? n and v were the size of snapshot validators